src/plugins/links/public/embeddable/links_embeddable.tsx

```diff
@@ -6,152 +6,228 @@
+import { openEditorFlyout } from '../editor/open_editor_flyout';
+
+import '../components/links_component.scss';
+import { checkForDuplicateTitle, linksClient } from '../content_management';
+import { resolveLinks } from '../lib/resolve_links';
+import {
+  deserializeLinksSavedObject,
+  linksSerializeStateIsByReference,
+} from '../lib/deserialize_from_library';
+
+export const LinksContext = createContext<LinksApi | null>(null);
+
+export const getLinksEmbeddableFactory = () => {
+  const linksEmbeddableFactory: ReactEmbeddableFactory<
+    LinksSerializedState,
+    LinksRuntimeState,
+    LinksApi
+  > = {
+    type: CONTENT_ID,
+    deserializeState: async (serializedState) => {
+      // Clone the state to avoid an object not extensible error when injecting references
+      const state = cloneDeep(serializedState.rawState);
+      const { title, description } = serializedState.rawState;
+
+      if (linksSerializeStateIsByReference(state)) {
+        const attributes = await deserializeLinksSavedObject(state);
+        return {
+          ...attributes,
+          title,
+          description,
+        };
+      }
+
+      const { attributes: attributesWithInjectedIds } = injectReferences({
+        attributes: state.attributes,
+        references: serializedState.references ?? [],
+      });
+
+      const resolvedLinks = await resolveLinks(attributesWithInjectedIds.links ?? []);
+
+      return {
+        title,
+        description,
+        links: resolvedLinks,
+        layout: attributesWithInjectedIds.layout,
+        defaultPanelTitle: attributesWithInjectedIds.title,
+        defaultPanelDescription: attributesWithInjectedIds.description,
+      };
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+      const { linksApi, linksComparators, serializeLinks } = initializeLinks(state, parentApi);
+      const savedObjectId$ = new BehaviorSubject(state.savedObjectId);
+
+      const serializeState = async (): Promise<SerializedPanelState<LinksSerializedState>> => {
+        const { attributes, references } = serializeLinks();
+        if (savedObjectId$.value !== undefined) {
+          const linksByReferenceState: LinksByReferenceSerializedState = {
+            savedObjectId: savedObjectId$.value,
+            ...serializeTitles(),
+          };
+
+          return { rawState: linksByReferenceState, references: [] };
+        }
+
+        const linksByValueState: LinksByValueSerializedState = {
+          attributes,
+          ...serializeTitles(),
+        };
+        return { rawState: linksByValueState, references };
+      };
+
+      const api = buildApi(
+        {
+          ...titlesApi,
+          ...linksApi,
+          libraryId$: savedObjectId$,
+          savedObjectId$,
+          getTypeDisplayName: () => APP_NAME,
+          getTypeDisplayNameLowerCase: () => 'links',
```

Comment:
`getTypeDisplayName` should be enough, right? Just make that return lowercase `links` rather than `Links`. That way it is changed in the "Add to library" action, too (which should be "Save links" and not "Save Links"):

![image](https://github.com/elastic/kibana/assets/8698078/285eddd1-72b0-48ca-9f03-a6be08ec718c)

This would be more consistent to how other embeddables do it. Also, this should be passed through `i18n`